app/server/api/vendor_payout_api.py

```diff
@@ -0,0 +1,177 @@
+from flask import Blueprint, request, send_file
+from flask.views import MethodView
+from flask import g
+
+from server.models.transfer_account import TransferAccount, TransferAccountType
+from server.models.credit_transfer import CreditTransfer
+from server.utils.credit_transfer import make_payment_transfer
+from server.utils.transfer_enums import TransferModeEnum
+from server.utils.auth import requires_auth
+from server import db
+
+import json
+import csv
+import io
+import codecs
+from decimal import Decimal
+
+vendor_payout = Blueprint('vendor_payout', __name__)
+
+class GetVendorPayoutAPI(MethodView):
+    @requires_auth(allowed_roles={'ADMIN': 'sempoadmin'})
+    def post(self):
+        # Process post data
+        post_data = request.get_json()
+        account_ids = []
+        if post_data:
+            account_ids = post_data.get('accounts', [])
+
+        if not isinstance(account_ids, list):
+            raise Exception('accounts parameter expects a list')
+        if account_ids:
+            vendors = db.session.query(TransferAccount).filter(TransferAccount.account_type != TransferAccountType.FLOAT).filter(TransferAccount.id.in_(account_ids)).all()
+            for vendor in vendors:
+                if not vendor.is_vendor:
+                    raise Exception(f'Transfer account with id {vendor.id} not a vendor account. Please only IDs of vendor accounts')
```

Comment:
When I create an account on the dashboard, this doesn't work as `is_vendor` on the transfer account is set to False. It seems with the update to how we handle role assignment it doesn't handle correctly anymore

Reply:
Changed to the users' is_vendor, as per our chat